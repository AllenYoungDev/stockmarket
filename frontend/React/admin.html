<html>
<head>
<style>
label:after { content: ':'; }
.top-bar { font-size: 100%; display:grid; grid-template-columns: 33% 34% 33%; margin: 0; padding: 1% 0.5% 1% 0; background: rgb(0,0,0); color: rgb(255,255,255); font-family: Arial, Helvetica, sans-serif; }
.top-bar-left-column { grid-row: 1; grid-column: 1; text-align: left; }
.top-bar-center-column { grid-row: 1; grid-column: 2; text-align: center; font-weight: bold;}
.top-bar-right-column { grid-row: 1; grid-column: 3; text-align: right; }
.user-list-section { margin: 0; padding: 0.5% 0 0.25% 0; background-color: #737CA1;color: rgb(250,250,250); }
.user-list-section-heading { text-align: center; font-size: 300%; font-family: Arial, Helvetica, sans-serif; font-weight: bold; margin: 0; }
.transaction-list-section { margin: 0; background-color: #FFFFFF;color: #36454F; }
.transaction-list-section-heading { text-align: center; font-size: 300%; font-family: Arial, Helvetica, sans-serif; font-weight: bold; margin: 0.5% 0 0 0; }

.user-list-div-display { text-align: center; margin: 2% 0 0 0;font-size: 120%; }
.user-list-table { color: rgb(250,250,250); text-align: center; margin: 0px auto; border-spacing: 1em 0em;}
.user-list-p-navigation { text-align: center; margin: 2% 0 0 0;font-size: 120%; }
.user-list-form { text-align: right; }

.transaction-list-div-display { text-align: center; margin: 2% 0 0 0;font-size: 120%; }
.transaction-list-table { color: #36454F; text-align: center; margin: 0px auto; border-spacing: 1em 0em;}
.transaction-list-p-navigation { text-align: center; margin: 2% 0 0 0;font-size: 120%; }
.transaction-list-form { text-align: right; }

.footer   {font-size: 100%; margin: 0; padding: 0.25% 0.5% 0.25% 0; background: rgb(0,0,0); color: rgb(255,255,255); font-family: Arial, Helvetica, sans-serif; text-align: center;}
</style>
</head>
<body style="margin: 0; background-color: #FFFFFF;color: #36454F;">

<div id="admin-account-page"></div>


<script src="https://unpkg.com/react@18/umd/react.development.js" crossorigin></script>
<script src="https://unpkg.com/react-dom@18/umd/react-dom.development.js" crossorigin></script>

<script type="module" src="admin.render.js"></script>

<noscript>Your browser must support JavaScript.</noscript>
</body>
</html>