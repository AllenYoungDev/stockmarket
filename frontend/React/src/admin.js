'use strict';

//import React, { StrictMode } from "react";
//import { createRoot } from 'react-dom/client';

import './settings/ServerUrls.js';
import './settings/dbMaximumNumberOfRowsReturned.js';
import './pageNavigationControlsGenerator.js';
import {fetchAccessTokenValidity} from './utilities/fetchAccessTokenValidity.js';
import {fetchTotalNumberOfUsers} from './utilities/fetchTotalNumberOfUsers.js';
import {fetchUserList} from './utilities/fetchUserList.js';
import {fetchTotalNumberOfTransactions} from './utilities/fetchTotalNumberOfTransactions.js';
import {fetchTransactionHistory} from './utilities/fetchTransactionHistory.js';
import {convertEpochTimeToLocalTimeString} from './utilities/convertEpochTimeToLocalTimeString.js';
import {handleLogoutOnClick} from './utilities/handleLogoutOnClick.js';
import {footer} from './footer.js';

/* *************************************************************************************************************
AdminAccountPage React component
************************************************************************************************************** */
export function AdminAccountPage() {
  const [loginState, setLoginState] = React.useState('');
    //loginState can be 'fetch-error', 'no-login', 'admin-login', and 'non-admin-login'.

  // /fetchAccessTokenValidity route requestwindow.
  // If access token is valid, admin cookie check and retrieval on client
  React.useEffect(() => {
    let ignore = false;
    setLoginState('');

    fetchAccessTokenValidity(setLoginState, ignore);

    return () => {
      ignore = true;
    }
  }, [setLoginState]);

  console.log(`AdminAccountPage() cookies on client:  ${document.cookie}.`)

  return (
    <React.Fragment>
    <div className="top-bar">
    <div className="top-bar-left-column"><a href={window.frontendServerUrl} style={{textDecoration: 'none', color: 'rgb(255,255,255)'}}>Allen Young's Stockmarket Demo</a></div>
    <div className="top-bar-center-column">Admin Page</div>
    <div className="top-bar-right-column">
    <MyAccountLogout loginState={loginState} />
    </div>
    </div>
    
    <section className="user-list-section">
    <h1 className="user-list-section-heading">User List</h1>
    <UserList loginState={loginState} />
    </section>

    <section className="transaction-list-section">
    <h1 className="transaction-list-section-heading">Transaction List</h1>  
    <TransactionList loginState={loginState} />
    </section>

    {footer()}
    </React.Fragment>
  );
}

/* *************************************************************************************************************
MyAccountLogout React component
************************************************************************************************************** */

export function MyAccountLogout({loginState}) {
  switch (loginState) {
    case 'fetch-error':
      return <div>Network error.  Please retry later.</div>;
    case 'no-login':
      return (
        <React.Fragment>
        <a href={window.frontendServerUrl + "/registration.html"} style={{textDecoration:'none', color: 'rgb(255,255,255)'}}>
          Register</a> |&nbsp;
        <a href={window.frontendServerUrl + "/login.html"} style={{textDecoration:'none', color: 'rgb(255,255,255)'}}>Login</a>
        </React.Fragment>
        );    
    case 'admin-login':
      return (
        <React.Fragment>
        <a href={window.frontendServerUrl + "/account.html"} style={{textDecoration:'none', color: 'rgb(255,255,255)'}}>My Account</a> |&nbsp;
        <a href="" onClick={handleLogoutOnClick} style={{textDecoration:'none', color: 'rgb(255,255,255)'}}>Logout</a>
        </React.Fragment>
        );
    case 'non-admin-login':
      return (
        <React.Fragment>
        <a href={window.frontendServerUrl + "/account.html"} style={{textDecoration:'none', color: 'rgb(255,255,255)'}}>My Account</a> |&nbsp;
        <a href="" onClick={handleLogoutOnClick} style={{textDecoration:'none', color: 'rgb(255,255,255)'}}>Logout</a>
        </React.Fragment>
        );
    default:
      /* The following is for doing better or smooth rendering. Refer to the "Better rendering methods"
         section in the developer's journal.  */
      if (document.cookie.includes('accessToken=')) {
        if (document.cookie.includes('admin=true')) {
          return (
            <React.Fragment>
            <a href={window.frontendServerUrl + "/account.html"} style={{textDecoration:'none', color: 'rgb(255,255,255)'}}>My Account</a> |&nbsp;
            <a href="" onClick={handleLogoutOnClick} style={{textDecoration:'none', color: 'rgb(255,255,255)'}}>Logout</a>
            </React.Fragment>
            );
        } else {
          return (
            <React.Fragment>
            <a href={window.frontendServerUrl + "/account.html"} style={{textDecoration:'none', color: 'rgb(255,255,255)'}}>My Account</a> |&nbsp;
            <a href="" onClick={handleLogoutOnClick} style={{textDecoration:'none', color: 'rgb(255,255,255)'}}>Logout</a>
            </React.Fragment>
            );
        }
      } else {
        return (
          <React.Fragment>
          <a href={window.frontendServerUrl + "/registration.html"} style={{textDecoration:'none', color: 'rgb(255,255,255)'}}>
            Register</a> |&nbsp;
          <a href={window.frontendServerUrl + "/login.html"} style={{textDecoration:'none', color: 'rgb(255,255,255)'}}>Login</a>
          </React.Fragment>
          );    
      }
  }
}


/* *************************************************************************************************************
UserList React component
************************************************************************************************************** */

export function UserList({loginState}) {
  const userListPageNumberInputRef = React.useRef(null);

  const [numberOfUsers, setNumberOfUsers] = React.useState(0);
  const [pageNumber, setPageNumber] = React.useState(1);
  const [userListPageData, setUserListPageData] = React.useState([]);

  // /GetTotalNumberOfUsers request.
  React.useEffect(() => {
    let ignore = false;
    setNumberOfUsers(0);

    fetchTotalNumberOfUsers(setNumberOfUsers, ignore);

    return () => {
      ignore = true;
    }
  }, [setNumberOfUsers]);

  // /GetUserList/:pageNumber request.
  React.useEffect(() => {
    let ignore = false;
    setUserListPageData([]);

    fetchUserList(pageNumber, setUserListPageData, ignore);

    return () => {
      ignore = true;
    }
  }, [pageNumber, setUserListPageData]);

  function handlePageNumberOnInput(event) {
    event.stopPropagation();
    if (!(event.target.validity.valid)) {
      event.target.value='';
    }
  }

  function handleGoToButtonOnClick(event) {
    event.stopPropagation();

    let userInputPageNumber = userListPageNumberInputRef.current.value.trim();
    if (userInputPageNumber == pageNumber || userInputPageNumber === '' || userInputPageNumber < 1 || 
      userInputPageNumber > numberOfUsers) {return;}

    setPageNumber(userInputPageNumber);
  }

  function handleNavigationLinkOnClick(event, navigationPageNumber) {
    event.preventDefault();
    event.stopPropagation();

    if (navigationPageNumber === 0) {return;}

    if (pageNumber !== navigationPageNumber) {
      setPageNumber(navigationPageNumber);
    }
  }

  if (loginState === 'admin-login') {
    console.log(`UserList(), loginState === 'admin-login', userListPageData:  ${JSON.stringify(userListPageData)}.`)

    const listItems = userListPageData.map(userData => {
      return (
        <tr key={userData["ID"]}>
        <td key={userData["ID"] + "_0"}>{userData["ID"]}</td>
        <td key={userData["ID"] + "_1"}>{userData["First name"]}</td>
        <td key={userData["ID"] + "_2"}>{userData["Last name"]}</td>
        <td key={userData["ID"] + "_3"}>{userData["Email address"]}</td>
        <td key={userData["ID"] + "_4"}>{userData["Email address verified"]}</td>
        <td key={userData["ID"] + "_5"}>{userData["Phone number"]}</td>
        <td key={userData["ID"] + "_6"}>{userData["Admin"]}</td>
        <td key={userData["ID"] + "_7"}>{userData["User deleted"]}</td>
        </tr>
      );
    });

    console.log(`UserList(), loginState === 'admin-login', listItems:  ${listItems}.`)

    var totalNumberOfPages = Math.ceil(numberOfUsers / window.dbMaximumNumberOfRowsReturned);

    const navigationControls = window.generatePageNavigationControls(totalNumberOfPages, pageNumber, 
      handleNavigationLinkOnClick, 'rgb(250,250,250)');

    return (
      <React.Fragment>
        <div className="user-list-div-display">
        <table className="user-list-table">
          <thead>
              <tr>
                  <th>User primary key</th>
                  <th>First name</th>
                  <th>Last name</th>
                  <th>Email address</th>
                  <th>Email address verified</th>
                  <th>Phone number</th>
                  <th>Admin</th>
                  <th>User deleted</th>
              </tr>
          </thead>
          <tbody>
                  {listItems}
          </tbody>
        </table>          
        </div>
        <p className="user-list-p-navigation">{navigationControls}</p>

        <form className="user-list-form">
          <label htmlFor="user_list_page_number">Page</label>
          <input type="number" id="user_list_page_number" name="user_list_page_number" size="5" min="1" max={totalNumberOfPages} step="1" onInput={handlePageNumberOnInput} ref={userListPageNumberInputRef} />
          <input type="button" value="Go to" onClick={handleGoToButtonOnClick} />
        </form>
        </React.Fragment>
    );
  } else {
    return (
      <React.Fragment>
        <div className="user-list-div-display">No records to show.</div>
        <p className="user-list-p-navigation">&lt; 1 2 3 â€¦ (last page number) &gt;</p>

        <form className="user-list-form">
          <label htmlFor="user_list_page_number">Page</label>
          <input disabled type="number" id="user_list_page_number" name="user_list_page_number" size="5" />
          <input disabled type="button" value="Go to" />
        </form>
      </React.Fragment>
    );    
  }
}


/* *************************************************************************************************************
TransactionList React component
************************************************************************************************************** */

export function TransactionList({loginState}) {
  const transactionListPageNumberInputRef = React.useRef(null);

  const [numberOfTransactions, setNumberOfTransactions] = React.useState(0);
  const [pageNumber, setPageNumber] = React.useState(1);
  const [transactionListPageData, setTransactionListPageData] = React.useState([]);

  // /GetTotalNumberOfTransactions request.
  React.useEffect(() => {
    let ignore = false;
    setNumberOfTransactions(0);

    fetchTotalNumberOfTransactions(setNumberOfTransactions, ignore);

    return () => {
      ignore = true;
    }
  }, [setNumberOfTransactions]);

  // /GetTransactionHistory/:pageNumber request.
  React.useEffect(() => {
    let ignore = false;
    setTransactionListPageData([]);

    fetchTransactionHistory(pageNumber, setTransactionListPageData, ignore);

    return () => {
      ignore = true;
    }
  }, [pageNumber, setTransactionListPageData]);

  function handlePageNumberOnInput(event) {
    event.stopPropagation();
    if (!(event.target.validity.valid)) {
      event.target.value='';
    }
  }

  function handleGoToButtonOnClick(event) {
    event.stopPropagation();

    let userInputPageNumber = transactionListPageNumberInputRef.current.value.trim();
    if (userInputPageNumber == pageNumber || userInputPageNumber === '' || userInputPageNumber < 1 || 
      userInputPageNumber > numberOfTransactions) {return;}

    setPageNumber(userInputPageNumber);
  }

  function handleNavigationLinkOnClick(event, navigationPageNumber) {
    event.preventDefault();
    event.stopPropagation();

    if (navigationPageNumber === 0) {return;}

    if (pageNumber !== navigationPageNumber) {
      setPageNumber(navigationPageNumber);
    }
  }

  if (loginState === 'admin-login') {
    console.log(`TransactionList(), loginState === 'admin-login', transactionListPageData:  ${JSON.stringify(transactionListPageData)}.`)

    const listItems = transactionListPageData.map(userData => {
      return (
        <tr key={userData["Primary key"]}>
        <td key={userData["Primary key"] + "_1"}>{userData["User primary key"]}</td>
        <td key={userData["Primary key"] + "_2"}>{userData["Stock stats table entry primary key"]}</td>
        <td key={userData["Primary key"] + "_3"}>{convertEpochTimeToLocalTimeString(userData["Transaction start datetime"])}</td>
        <td key={userData["Primary key"] + "_4"}>{convertEpochTimeToLocalTimeString(userData["Transaction end datetime"])}</td>
        <td key={userData["Primary key"] + "_5"}>{userData["PayPal transaction (order) ID"]}</td>
        <td key={userData["Primary key"] + "_6"}>{userData["Company stock transaction ID"]}</td>
        <td key={userData["Primary key"] + "_7"}>{userData["Number of shares"]}</td>
        <td key={userData["Primary key"] + "_8"}>{userData["Payment processing initiated"]}</td>
        <td key={userData["Primary key"] + "_9"}>{userData["Payment processing completed"]}</td>
        <td key={userData["Primary key"] + "_10"}>{userData["Payment processing status"]}</td>
        </tr>
      );
    });

    console.log(`TransactionList(), loginState === 'admin-login', listItems:  ${listItems}.`)

    var totalNumberOfPages = Math.ceil(numberOfTransactions / window.dbMaximumNumberOfRowsReturned);

    const navigationControls = window.generatePageNavigationControls(totalNumberOfPages, pageNumber, 
      handleNavigationLinkOnClick, '#36454F');

    return (
      <React.Fragment>
        <div className="transaction-list-div-display">
        <table className="transaction-list-table">
          <thead>
              <tr>
                  <th>User primary key</th>
                  <th>Stock stats table entry primary key</th>
                  <th>Transaction start datetime</th>
                  <th>Transaction end datetime</th>
                  <th>PayPal transaction (order) ID</th>
                  <th>Company stock transaction ID</th>
                  <th>Number of shares</th>
                  <th>Payment processing initiated</th>
                  <th>Payment processing completed</th>
                  <th>Payment processing status</th>
              </tr>
          </thead>
          <tbody>
                  {listItems}
          </tbody>
        </table>          
        </div>
        <p className="transaction-list-p-navigation">{navigationControls}</p>

        <form className="transaction-list-form">
          <label htmlFor="transaction_history_page_number">Page</label>
          <input type="number" id="transaction_history_page_number" name="transaction_history_page_number" size="5" min="1" max={totalNumberOfPages} step="1" onInput={handlePageNumberOnInput} ref={transactionListPageNumberInputRef} />
          <input type="button" value="Go to" onClick={handleGoToButtonOnClick} />
        </form>
        </React.Fragment>
    );
  } else {
    return (
      <React.Fragment>
        <div className="transaction-list-div-display">No records to show.</div>
        <p className="transaction-list-p-navigation">&lt; 1 2 3 â€¦ (last page number) &gt;</p>

        <form className="transaction-list-form">
          <label htmlFor="transaction_history_page_number">Page</label>
          <input disabled type="number" id="transaction_history_page_number" name="transaction_history_page_number" size="5" />
          <input disabled type="button" value="Go to" />
        </form>
      </React.Fragment>
    );    
  }
}

