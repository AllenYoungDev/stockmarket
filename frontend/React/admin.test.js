/**
 * @jest-environment jsdom
 */

//import React from "react";
//import ReactDOM from 'react-dom/client';

import renderer from 'react-test-renderer';
import {AdminAccountPage, MyAccountLogout, UserList, TransactionList} from './admin.js';

jest.mock('./utilities/fetchAccessTokenValidity');
jest.mock('./utilities/fetchTotalNumberOfUsers');
jest.mock('./utilities/fetchUserList');
jest.mock('./utilities/fetchTotalNumberOfTransactions');
jest.mock('./utilities/fetchTransactionHistory');

import {fetchAccessTokenValidity} from './utilities/fetchAccessTokenValidity.js';
import {fetchTotalNumberOfUsers} from './utilities/fetchTotalNumberOfUsers.js';
import {fetchUserList} from './utilities/fetchUserList.js';
import {fetchTotalNumberOfTransactions} from './utilities/fetchTotalNumberOfTransactions.js';
import {fetchTransactionHistory} from './utilities/fetchTransactionHistory.js';

(async () => {
  if (typeof React === 'undefined') {
    globalThis.React = await import("react");
  }
  if (typeof ReactDOM === 'undefined') {
    globalThis.ReactDOM = await import("react-dom/client");
  }
})();

var loginState = '';

async function adminAccountPageJestSnapshotTest(loginStateArg) {
  let testRenderer;
  let testInstance;
  let tree;

  loginState = loginStateArg;

  //renderer.create() is wrapped in renderer.act() for executing useEffect() calls in <HomePage />.
  //https://github.com/facebook/react/issues/15321
  await renderer.act(async () => {
    testRenderer = renderer.create(
      <AdminAccountPage />,
    );
  });

  testInstance = testRenderer.root;
  expect(testInstance.findByType(MyAccountLogout).props.loginState).toBe(loginStateArg);
  expect(testInstance.findByType(UserList).props.loginState).toBe(loginStateArg);
  expect(testInstance.findByType(TransactionList).props.loginState).toBe(loginStateArg);

  tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
}

it('React app Jest snapshot testing for Allen Young\'s Stockmarket Demo admin account page', async () => {

  fetchAccessTokenValidity.mockImplementation((setLoginState, ignore) => {
    console.log('fetchAccessTokenValidity() mock called!');
    setLoginState(loginState);
  });
  fetchTotalNumberOfUsers.mockImplementation((setNumberOfUsers, ignore) => {
    console.log('fetchTotalNumberOfUsers() mock called!');
    setNumberOfUsers(0);
  });
  fetchTotalNumberOfTransactions.mockImplementation((setNumberOfTransactions, ignore) => {
    console.log('fetchTotalNumberOfTransactions() mock called!');
    setNumberOfTransactions(0);
  });
  fetchUserList.mockImplementation((pageNumber, setUserListPageData, ignore) => {
    console.log('fetchUserList() mock called!');
    setUserListPageData([]);
  });
  fetchTransactionHistory.mockImplementation((pageNumber, setTransactionListPageData, ignore) => {
    console.log('fetchTransactionHistory() mock called!');
    setTransactionListPageData([]);
  });

  /*************************************************************************************************
  No-login mock Jest snapshot test
  **************************************************************************************************/
  await adminAccountPageJestSnapshotTest('no-login');

  /*************************************************************************************************
  Non-admin-user-login mock Jest snapshot test
  **************************************************************************************************/
  await adminAccountPageJestSnapshotTest('non-admin-login');

  /*************************************************************************************************
  Admin-user-login mock Jest snapshot test
  **************************************************************************************************/
  await adminAccountPageJestSnapshotTest('admin-login');

  /*************************************************************************************************
  Fetch-error mock Jest snapshot test
  **************************************************************************************************/
  await adminAccountPageJestSnapshotTest('fetch-error');
});