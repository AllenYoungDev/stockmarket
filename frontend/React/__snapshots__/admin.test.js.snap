// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo admin account page 1`] = `
[
  <div
    className="top-bar"
  >
    <div
      className="top-bar-left-column"
    >
      <a
        href="http://localhost"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Allen Young's Stockmarket Demo
      </a>
    </div>
    <div
      className="top-bar-center-column"
    >
      Admin Page
    </div>
    <div
      className="top-bar-right-column"
    >
      <a
        href="http://localhost/registration.html"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Register
      </a>
       | 
      <a
        href="http://localhost/login.html"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Login
      </a>
    </div>
  </div>,
  <section
    className="user-list-section"
  >
    <h1
      className="user-list-section-heading"
    >
      User List
    </h1>
    <div
      className="company-stock-transaction-history-div-display"
    >
      No records to show.
    </div>
    <p
      className="company-stock-transaction-history-p-navigation"
    >
      &lt; 1 2 3 … (last page number) &gt;
    </p>
    <form
      className="company-stock-transaction-history-form"
    >
      <label
        htmlFor="page_number"
      >
        Page
      </label>
      <input
        disabled={true}
        id="page_number"
        name="page_number"
        size="5"
        type="text"
      />
      <input
        disabled={true}
        type="button"
        value="Go to"
      />
    </form>
  </section>,
  <section
    className="transaction-list-section"
  >
    <h1
      className="transaction-list-section-heading"
    >
      Transaction List
    </h1>
    <div
      className="company-stock-transaction-history-div-display"
    >
      No records to show.
    </div>
    <p
      className="company-stock-transaction-history-p-navigation"
    >
      &lt; 1 2 3 … (last page number) &gt;
    </p>
    <form
      className="company-stock-transaction-history-form"
    >
      <label
        htmlFor="page_number"
      >
        Page
      </label>
      <input
        disabled={true}
        id="page_number"
        name="page_number"
        size="5"
        type="text"
      />
      <input
        disabled={true}
        type="button"
        value="Go to"
      />
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      © 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo admin account page 2`] = `
[
  <div
    className="top-bar"
  >
    <div
      className="top-bar-left-column"
    >
      <a
        href="http://localhost"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Allen Young's Stockmarket Demo
      </a>
    </div>
    <div
      className="top-bar-center-column"
    >
      Admin Page
    </div>
    <div
      className="top-bar-right-column"
    >
      <a
        href="http://localhost/account.html"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        My Account
      </a>
       | 
      <a
        href=""
        onClick={[Function]}
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Logout
      </a>
    </div>
  </div>,
  <section
    className="user-list-section"
  >
    <h1
      className="user-list-section-heading"
    >
      User List
    </h1>
    <div
      className="company-stock-transaction-history-div-display"
    >
      No records to show.
    </div>
    <p
      className="company-stock-transaction-history-p-navigation"
    >
      &lt; 1 2 3 … (last page number) &gt;
    </p>
    <form
      className="company-stock-transaction-history-form"
    >
      <label
        htmlFor="page_number"
      >
        Page
      </label>
      <input
        disabled={true}
        id="page_number"
        name="page_number"
        size="5"
        type="text"
      />
      <input
        disabled={true}
        type="button"
        value="Go to"
      />
    </form>
  </section>,
  <section
    className="transaction-list-section"
  >
    <h1
      className="transaction-list-section-heading"
    >
      Transaction List
    </h1>
    <div
      className="company-stock-transaction-history-div-display"
    >
      No records to show.
    </div>
    <p
      className="company-stock-transaction-history-p-navigation"
    >
      &lt; 1 2 3 … (last page number) &gt;
    </p>
    <form
      className="company-stock-transaction-history-form"
    >
      <label
        htmlFor="page_number"
      >
        Page
      </label>
      <input
        disabled={true}
        id="page_number"
        name="page_number"
        size="5"
        type="text"
      />
      <input
        disabled={true}
        type="button"
        value="Go to"
      />
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      © 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo admin account page 3`] = `
[
  <div
    className="top-bar"
  >
    <div
      className="top-bar-left-column"
    >
      <a
        href="http://localhost"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Allen Young's Stockmarket Demo
      </a>
    </div>
    <div
      className="top-bar-center-column"
    >
      Admin Page
    </div>
    <div
      className="top-bar-right-column"
    >
      <a
        href="http://localhost/account.html"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        My Account
      </a>
       | 
      <a
        href=""
        onClick={[Function]}
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Logout
      </a>
    </div>
  </div>,
  <section
    className="user-list-section"
  >
    <h1
      className="user-list-section-heading"
    >
      User List
    </h1>
    <div
      className="user-list-div-display"
    >
      <table
        className="user-list-table"
      >
        <thead>
          <tr>
            <th>
              User primary key
            </th>
            <th>
              First name
            </th>
            <th>
              Last name
            </th>
            <th>
              Email address
            </th>
            <th>
              Email address verified
            </th>
            <th>
              Phone number
            </th>
            <th>
              Admin
            </th>
            <th>
              User deleted
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <p
      className="user-list-p-navigation"
    >
      <a
        href=""
        onClick={[Function]}
        style={
          {
            "color": "rgb(250,250,250)",
            "textDecoration": "none",
          }
        }
      >
        &lt;
      </a>
      <span>
         
      </span>
      <a
        href=""
        onClick={[Function]}
        style={
          {
            "color": "rgb(250,250,250)",
            "textDecoration": "none",
          }
        }
      >
        &gt;
      </a>
    </p>
    <form
      className="user-list-form"
    >
      <label
        htmlFor="page_number"
      >
        Page
      </label>
      <input
        id="page_number"
        max={0}
        min="1"
        name="page_number"
        onInput={[Function]}
        size="5"
        step="1"
        type="number"
      />
      <input
        onClick={[Function]}
        type="button"
        value="Go to"
      />
    </form>
  </section>,
  <section
    className="transaction-list-section"
  >
    <h1
      className="transaction-list-section-heading"
    >
      Transaction List
    </h1>
    <div
      className="transaction-list-div-display"
    >
      <table
        className="transaction-list-table"
      >
        <thead>
          <tr>
            <th>
              User primary key
            </th>
            <th>
              Stock stats table entry primary key
            </th>
            <th>
              Transaction start datetime
            </th>
            <th>
              Transaction end datetime
            </th>
            <th>
              PayPal transaction (order) ID
            </th>
            <th>
              Company stock transaction ID
            </th>
            <th>
              Number of shares
            </th>
            <th>
              Payment processing initiated
            </th>
            <th>
              Payment processing completed
            </th>
            <th>
              Payment processing status
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <p
      className="transaction-list-p-navigation"
    >
      <a
        href=""
        onClick={[Function]}
        style={
          {
            "color": "#36454F",
            "textDecoration": "none",
          }
        }
      >
        &lt;
      </a>
      <span>
         
      </span>
      <a
        href=""
        onClick={[Function]}
        style={
          {
            "color": "#36454F",
            "textDecoration": "none",
          }
        }
      >
        &gt;
      </a>
    </p>
    <form
      className="transaction-list-form"
    >
      <label
        htmlFor="page_number"
      >
        Page
      </label>
      <input
        id="page_number"
        max={0}
        min="1"
        name="page_number"
        onInput={[Function]}
        size="5"
        step="1"
        type="number"
      />
      <input
        onClick={[Function]}
        type="button"
        value="Go to"
      />
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      © 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo admin account page 4`] = `
[
  <div
    className="top-bar"
  >
    <div
      className="top-bar-left-column"
    >
      <a
        href="http://localhost"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Allen Young's Stockmarket Demo
      </a>
    </div>
    <div
      className="top-bar-center-column"
    >
      Admin Page
    </div>
    <div
      className="top-bar-right-column"
    >
      <div>
        Network error.  Please retry later.
      </div>
    </div>
  </div>,
  <section
    className="user-list-section"
  >
    <h1
      className="user-list-section-heading"
    >
      User List
    </h1>
    <div
      className="company-stock-transaction-history-div-display"
    >
      No records to show.
    </div>
    <p
      className="company-stock-transaction-history-p-navigation"
    >
      &lt; 1 2 3 … (last page number) &gt;
    </p>
    <form
      className="company-stock-transaction-history-form"
    >
      <label
        htmlFor="page_number"
      >
        Page
      </label>
      <input
        disabled={true}
        id="page_number"
        name="page_number"
        size="5"
        type="text"
      />
      <input
        disabled={true}
        type="button"
        value="Go to"
      />
    </form>
  </section>,
  <section
    className="transaction-list-section"
  >
    <h1
      className="transaction-list-section-heading"
    >
      Transaction List
    </h1>
    <div
      className="company-stock-transaction-history-div-display"
    >
      No records to show.
    </div>
    <p
      className="company-stock-transaction-history-p-navigation"
    >
      &lt; 1 2 3 … (last page number) &gt;
    </p>
    <form
      className="company-stock-transaction-history-form"
    >
      <label
        htmlFor="page_number"
      >
        Page
      </label>
      <input
        disabled={true}
        id="page_number"
        name="page_number"
        size="5"
        type="text"
      />
      <input
        disabled={true}
        type="button"
        value="Go to"
      />
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      © 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;
